### PR TITLE
Add documentation for TLS configuration of NLK

### PR DESCRIPTION
**Pull request by ciroque**
### Proposed changes

Provide documentation for configuring NLK to use TLS for communications with NGINX Plus.

This is just the documentation, the implementation will come along later.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [`CONTRIBUTING`](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/blob/main/CONTRIBUTING.md) document
- [X] If applicable, I have added tests that prove my fix is effective or that my feature works
- [X] If applicable, I have checked that any relevant tests pass after adding my changes
- [X] I have updated any relevant documentation ([`README.md`](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/blob/main/README.md) and [`CHANGELOG.md`](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/blob/main/CHANGELOG.md))

### CODE REVIEW

File: docs/tls/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Securing communication between NLK and NGINX Plus with TLS / mTLS
+
+<span style="font-size:xx-large; color: #FF3333; font-weight: bolder;">THIS FEATURE IS IN DEVELOPMENT. THIS NOTICE WILL BE REMOVED WHEN THE IMPLEMENTATION IS COMPLETE.</span>
```
Comment (4141done) [inline docs/tls/README.md:3]: There's a GitHub Flavored Markdown solution to this that also helps make it a littlemore visible:

```
> [!WARNING]  
> Critical content demanding immediate user attention due to potential risks.
```

File: docs/tls/KUBERNETES-SECRETS.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Kubernetes Secrets
+
+## Overview
+
+Kubernetes Secrets are used to provide the required certificates to NLK. There are two ways to create the Secrets:
+- Using `kubectl`
+- Using yaml files
+
+The filenames for the certificates created previously are required for both methods. The examples below assume the certificates were generated in `/tmp`
```
Comment (4141done) [inline docs/tls/KUBERNETES-SECRETS.md:9]: In this case "previously" isn't very helpful. Maybe link to the cert generation docs?

File: docs/tls/KUBERNETES-SECRETS.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+The filenames for the certificates created previously are required for both methods. The examples below assume the certificates were generated in `/tmp` 
+and follow the naming conventions in the documentation. 
+
+## Using `kubectl`
+
+The easiest way to create the secret is by using `kubectl`:
+
+```bash
+kubectl create secret tls -n nlk nlk-tls-{ca,server,client}-secret --cert=/tmp/{ca,server,client}.crt --key=/tmp/{ca,server,client}.key
```
Comment (4141done) [inline docs/tls/KUBERNETES-SECRETS.md:17]: My ignorance on `kubectl` but does `{ca, server, client}` actually use all those names to create seperate secrets, or are you trying to denote that you'll need to run the command once for each file?  If so, although it's annoying I'd suggest providing all the discrete commands.  If not - please ignore me.

File: docs/tls/KUBERNETES-SECRETS.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+type: kubernetes.io/tls
+```
+
+Note: While it is possible to generate the values for `tls.crt` and `tls.key` manually, the above yaml can be generated using the following command:
+
+```bash
+kubectl create secret tls -n nlk nlk-tls-{ca,server,client}-secret --cert=/tmp/{ca,server,client}.crt --key=/tmp/{ca,server,client}.key --dry-run=client -o yaml > {ca,server,client}-secret.yaml
+```
+
+**Warning: it is important that these files do not make their way into a public repository or other storage location where they can be accessed by unauthorized users.**
```
Comment (4141done) [inline docs/tls/KUBERNETES-SECRETS.md:41]: Same note that the warning with the icon can be used here.  Ref: https://github.com/orgs/community/discussions/16925

**Review by 4141done — APPROVED** [review]
A few small notes but I suspect some things will change with implementation. Further improvements can come from user feedback. Might want to consider not merging this until implementation is complete but you've also provided ample warning so up to you.

**Comment by ciroque** [comment]
Decided it would be worse to throw this in with the [implementation PR](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/pull/120). Happy hunting.

**Pull request closed without merging**